### DISCUSSION TITLE
the way relations is found throws an error when using Scout

### DISCUSSION THREAD

**Discussion by rabol** [GeorgeFourkas/ApiAutoPilot#12] [Q&A]
Hi

AAP throws an error on a model that I have included in Laravel Scout

the error is:
```
"message": "Call to undefined method App\\Models\\Recipe::trashed()",
```

and comes from the method ``` pushSoftDeleteMetadata() ``` in the Searchable.php file, line 346

I don't think it is a 'bug' in the scout code because as far as I can see, the ``` pushSoftDeleteMetadata() ```is only called when the soft delete is enabled, it's more the way that the AAP package is trying to find the relations on a model.

Any hint's on how to prevent this ?

**Comment by GeorgeFourkas** [comment]
Api Auto Pilot, does not currently support Searching with Scout. This feature will become available in the next major version. To avoid running to this issue you can use the built-in search endpoints. You can find more information in the documentaion website https://apiautopilot.info/guide/endpoints/available-endpoints.html#searching

**Reply by rabol** [reply]
It has nothing todo with searching.
the problem is that aap 'executes' all public methods on the model to find relations, and on of the 'scout' methods uses the ->trashed(), but before calling this method Scout checks for 'soft deletes so it is not a problem with scout, but aap

**Comment by rabol** [comment]
e.g this will also cause aap to fail

```
public function test(): void
{
    $this->trashed();
}
```
